sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/factory/AzureCredentialBuilderFactory.java

```diff
@@ -90,15 +108,34 @@ protected void configureService(T builder) {
     @Override
     protected void configureRetry(T builder) {
         RetryAware.Retry retry = getAzureProperties().getRetry();
-        if (retry != null && retry.getMaxAttempts() != null) {
+        if (retry == null) {
+            return;
+        }
+
+        if (retry.getMaxAttempts() != null) {
             builder.maxRetry(retry.getMaxAttempts());
-            // TODO (xiada): don't know how to specify the retryTimeout
-//            builder.retryTimeout()
         }
+        Function<Duration, Duration> retryTimeout = retryTimeout();
+        if (retryTimeout != null) {
+            builder.retryTimeout(retryTimeout);
+        }
+    }
+
+    /**
+     * Default timeout implementation
+     * @return Timeout function
+     */
+    protected Function<Duration, Duration> retryTimeout() {
+        RetryAware.Retry retry = getAzureProperties().getRetry();
+        if (retry == null || retry.getTimeout() == null) {
+            return null;
+        }
+        return timeout -> retry.getTimeout();
```

Comment:
The timeout is about the maximum duration to wait for completion of a single attempt, whether the initial attempt or a retry.